checker-qual/src/main/java/module-info.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * This module contains annotations (type qualifiers) that a programmer writes to specify Java code
+ * for type-checking by the Checker Framework.
+ */
+module org.checkerframework.checker.qual {
+  // javadoc-only dependencies
+  requires static java.compiler;
```

Comment:
If this is really for Javadoc only, wouldn't it be cleaner to remove those `requires static` statements and replace them by `--add-reads` options when invoking the `javadoc` tool?

Reply:
Technically, you can even skip the `--add-reads` because the javadoc phase does not read the module descriptor (due to circular dependencies which are not allowed with JPMS). 

I mainly added them to be as explicit as possible with the dependency declarations. For example, IntelliJ would flag even more references if these dependencies were not specified. I don't know how well IDEs are able to include phase-specific build-flags.

Do you have any issues with these declarations? As far as I know, `requires static` should not affect dependent modules.

Reply:
No issue that I'm aware of for now. I thought that those declarations are a risk, because the code could accidentally use those dependencies without warning from the compiler. The argument about being explicit can also be turned the other way around, i.e. they are not describing the reality if the code has no dependency to those modules. Anyway, this is not a major issue.